book/routing.rst

```diff
@@ -16,104 +16,10 @@ the change is simple.
-
-        class BlogController extends Controller
-        {
-            /**
-             * @Route("/blog/{slug}", name="blog_show")
-             */
-            public function showAction($slug)
-            {
-                // ...
-            }
-        }
-
-    .. code-block:: yaml
-
-        # app/config/routing.yml
-        blog_show:
-            path:      /blog/{slug}
-            defaults:  { _controller: AppBundle:Blog:show }
-
-    .. code-block:: xml
-
-        <!-- app/config/routing.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <routes xmlns="http://symfony.com/schema/routing"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xsi:schemaLocation="http://symfony.com/schema/routing
-                http://symfony.com/schema/routing/routing-1.0.xsd">
-
-            <route id="blog_show" path="/blog/{slug}">
-                <default key="_controller">AppBundle:Blog:show</default>
-            </route>
-        </routes>
-
-    .. code-block:: php
-
-        // app/config/routing.php
-        use Symfony\Component\Routing\RouteCollection;
-        use Symfony\Component\Routing\Route;
-
-        $collection = new RouteCollection();
-        $collection->add('blog_show', new Route('/blog/{slug}', array(
-            '_controller' => 'AppBundle:Blog:show',
-        )));
-
-        return $collection;
-
-.. versionadded:: 2.2
-    The ``path`` option was introduced in Symfony 2.2, ``pattern`` is used
-    in older versions.
-
-The path defined by the ``blog_show`` route acts like ``/blog/*`` where
-the wildcard is given the name ``slug``. For the URL ``/blog/my-blog-post``,
-the ``slug`` variable gets a value of ``my-blog-post``, which is available
-for you to use in your controller (keep reading). The ``blog_show`` is the
-internal name of the route, which doesn't have any meaning yet and just needs
-to be unique. Later, you'll use it to generate URLs.
-
-If you don't want to use annotations, because you don't like them or because
-you don't want to depend on the SensioFrameworkExtraBundle, you can also use
-Yaml, XML or PHP. In these formats, the ``_controller`` parameter is a special
-key that tells Symfony which controller should be executed when a URL matches
-this route. The ``_controller`` string is called the
-:ref:`logical name <controller-string-syntax>`. It follows a pattern that
-points to a specific PHP class and method, in this case the
-``AppBundle\Controller\BlogController::showAction`` method.
-
-Congratulations! You've just created your first route and connected it to
-a controller. Now, when you visit ``/blog/my-post``, the ``showAction`` controller
-will be executed and the ``$slug`` variable will be equal to ``my-post``.
-
-This is the goal of the Symfony router: to map the URL of a request to a
-controller. Along the way, you'll learn all sorts of tricks that make mapping
-even the most complex URLs easy.
```

Comment:
Ah, but we could probably keep this section and delete *everything* else - this is the *heart* of this document :). If I understand it correctly, now the first routing example will be *very* far down on the page, after a lot of boring paragraphs (people skip this mostly and look for code blocks).

In other words, I think this should stay!